### PR TITLE
fast 'repipe url' command to construct concourse dashboard url

### PR DESCRIPTION
**Pull request by drnic**
Script is in https://github.com/cloudfoundry-community/mattermost-boshrelease already for a demo.

On OSX:

```
git clone https://github.com/cloudfoundry-community/mattermost-boshrelease
cd mattermost-boshrelease
ci/repipe url | xargs -L1 open
```

I'd really like `ci/repipe open` but couldn't get that working. Ideas?

### CODE REVIEW

File: repipe [hunk 1]
```diff
@@ -41,6 +41,20 @@ usage() {
     echo "validate          Validatei pipeline instead of set pipeline"
     echo "validate-strict   Validate pipeline with strict mode"
     echo "non-interactive   Run set-pipeline in non-interactive mode"
+    echo "url               Pipeline dashboard URL. Usage: ci/repipe url | xargs -L1 open"
+}
+
+show_pipeline_url() {
+	spruce merge --skip-eval pipeline.yml ${settings_file} > .deploy.yml
```
Comment (drnic) [inline repipe:64]: Using `--skip-eval` so that the command can be fast, and not slowed down by a dozen lookups to Vault.

File: repipe [hunk 2]
```diff
@@ -41,6 +41,20 @@ usage() {
...
+	concourse_url=$(spruce json .deploy.yml | jq -r ".meta.url")
+	team=$(spruce json .deploy.yml | jq -r ".meta.team // \"main\"")
+	pipeline=$(spruce merge --skip-eval \
```
Comment (drnic) [inline repipe:67]: Now need to perform `spruce merge` to get `meta.pipeline` but still be fast, so cherry-pick two required items

**Comment by jhunt** [comment]
I was able to get `./ci/repipe open` working for me, using the following patch to the version in mattermost-boshrelease:

```
diff --git a/ci/repipe b/ci/repipe
index 49538e5..1752b51 100755
--- a/ci/repipe
+++ b/ci/repipe
@@ -44,6 +44,22 @@ usage() {
     echo "url               Pipeline dashboard URL. Usage: ci/repipe url | xargs -L1 open"
 }
 
+open_pipeline() {
+	url=$(show_pipeline_url)
+	if [[ -x /usr/bin/open ]]; then
+		exec /usr/bin/open "$url"
+	else
+		echo "Sorry, but I was not able to automagically open"
+		echo "your Concourse Pipeline in the browser."
+		echo
+		echo "Here's a link you can click on, though:"
+		echo
+		echo "  $url"
+		echo
+		exit 0;
+	fi
+}
+
 show_pipeline_url() {
 	spruce merge --skip-eval pipeline.yml ${settings_file} > .deploy.yml
 	concourse_url=$(spruce json .deploy.yml | jq -r ".meta.url")
@@ -65,6 +81,7 @@ for arg do
         validate-strict|validate_strict) VALIDATE_PIPELINE="strict" ;;
         non-interactive|non_interactive) NON_INTERACTIVE="--non-interactive" ;;
         url) SHOW_PIPELINE_URL="yes" ;;
+        open) OPEN_PIPELINE=yes ;;
         help|-h|--help) usage; exit 0 ;;
         *) echo Invalid argument
             usage
@@ -89,6 +106,7 @@ echo >&2 "Using settings found in ${settings_file}"
 set -e
 trap "cleanup" QUIT TERM EXIT INT
 
+[[ -n ${OPEN_PIPELINE} ]]     && { open_pipeline;     exit 0; }
 [[ -n ${SHOW_PIPELINE_URL} ]] && { show_pipeline_url; exit 0; }
 
 spruce merge pipeline.yml ${settings_file} > .deploy.yml
```

**Review by jhunt — APPROVED** [review]
LGTM

**Pull request merged**